src/ToData.purs

```diff
@@ -44,110 +51,179 @@ class ToData :: Type -> Constraint
 class ToData a where
   toData :: a -> PlutusData
 
-class ToDataWithIndex :: Type -> Type -> Constraint
-class HasConstrIndices ci <= ToDataWithIndex a ci where
-  toDataWithIndex :: Proxy ci -> a -> PlutusData
+{- | A class which converts a type to its PlutusData representation with the help of a
```

Comment:
You need to prefix each line of doc-comment with `|`:

https://github.com/purescript/documentation/blob/master/language/Syntax.md#comments

> Note that, unlike Haskell, every line which should be considered documentation must start with a pipe. This allows you to do things like:

Reply:
Will do this in #417